xc/hotplug.ml

```diff
@@ -229,15 +229,20 @@ let run_hotplug_script device args =
     | Vbd _ -> !Xc_resources.vbd_script
     | Vif | Tap -> !Xc_resources.vif_script
     | _ -> failwith (Printf.sprintf "don't know how to run a hotplug script for: %s" kind) in
+  let backend_prefix = match device.backend.domid with
+    | 0 -> ""
+    | id -> Printf.sprintf "/local/domain/%d/" id
+  in
   let env = Array.concat [ Unix.environment (); [|
       "script=" ^ script;
       "XENBUS_TYPE=" ^ kind;
       "XENBUS_PATH=" ^ (Printf.sprintf "backend/%s/%d/%d" kind device.frontend.domid device.frontend.devid);
       "XENBUS_BASE_PATH=backend";
+      "XENBUS_PREFIX=" ^ backend_prefix;
       "INTERFACE=" ^ (Printf.sprintf "%s%d.%d" kind device.frontend.domid device.frontend.devid);
     |] ] in
   try
-    debug "Running hotplug script %s %s" script (String.concat " " args);
+    debug "Running hotplug script %s %s (%s)" script (String.concat " " args) (Array.to_list env |> String.concat " ");
```

Comment:
this looks like it will log all environment variables? you probably only want to log the things you added

Reply:
I should avoid logging Unix.environment (I do need all the other XENBUS* and INTERFACE values to be able to debug the script)